libdrgn/memory_reader.c

```diff
@@ -240,9 +240,9 @@ struct drgn_error *drgn_memory_reader_read(struct drgn_memory_reader *reader,
 		segment = drgn_memory_segment_tree_search_le(tree,
 							     &address).entry;
 		if (!segment || segment->address + segment->size <= address) {
-			return drgn_error_format(DRGN_ERROR_FAULT,
-						 "could not find memory segment containing 0x%" PRIx64,
-						 address);
+			return drgn_error_format_read(DRGN_ERROR_FAULT, address,
+						      "could not find memory segment containing 0x%" PRIx64,
+						      address);
```

Comment:
Duplicating the address between the message and the message is kind of icky. How about we make this `drgn_error_create_fault(const char *message, uint64_t address)` and have `string_builder_append_error()` format it specially as "message: address" like it does for OS errors?